### PR TITLE
Fix handling subscription with empty data

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because it fixes [#14834](https://github.com/saleor/saleor/issues/14834)
This should not break how any sync webhook work at the moment. The only change will be that if something with subscription was incorrect and payload generation resulted in empty data - it will be logged and returned as None instead of raising an Error. It will work the same as the event was not subscribable. 
Additionally some redundant lines from tests were removed. 

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by maarcingebala — APPROVED** [review]
Looks OK, but Black is failing, please check linters.

**Review by Air-t — APPROVED** [review]

**Review by korycins — APPROVED** [review]
Looks good 👍 
@jakubkuc  can you confirm, that you double-checked the logic that using the `create_delivery_for_subscription_sync_event(` implicitly and explicitly, and we don't have any unneeded try, except for `PaymentError`?

File: saleor/plugins/webhook/tasks.py [hunk 1]
```diff
@@ -164,12 +164,12 @@ def create_delivery_for_subscription_sync_event(
         app=webhook.app,
     )
     if not data:
-        # PaymentError is a temporary exception type. New type will be implemented
-        # in separate PR to ensure proper handling for all sync events.
-        # It was implemented when sync webhooks were handling payment events only.
-        raise PaymentError(
+        logger.info(
             f"No payload was generated with subscription for event: {event_type}"
```
Comment (8r2y5) [inline saleor/plugins/webhook/tasks.py:168]: Please use % string formatting when dealing with loggers

**Comment by jakubkuc** [comment]
@korycins Yes, I double-checked, this PaymentError was not handled anywhere apparently.

File: digital_contents/sample_image_as_digital_content.jpg
Comment (szczecha) [inline digital_contents/sample_image_as_digital_content.jpg:1]: These files should be omitted from PR

File: export_files/temp_file.csv
Comment (szczecha) [inline export_files/temp_file.csv:1]: These files should be omitted from PR

**Review by szczecha — APPROVED** [review]
@jakubkuc

File: saleor/plugins/webhook/tasks.py (hunk 1 above)
Comment (jakubkuc) [reply saleor/plugins/webhook/tasks.py]: done 👍🏻

File: export_files/temp_file.csv
Comment (jakubkuc) [reply export_files/temp_file.csv:1]: of course, it was a mistake. fixed

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**